docs/pjrt.md

```diff
@@ -0,0 +1,142 @@
+
+## Quickstart
+
+To start using PjRt with PyTorch/XLA, all you need to do is set the
+`PJRT_DEVICE` environment variable. If you're working on a TPU v2 or v3, keep
+reading to learn about the differences between TPU v2 and v3 and v4.
+
+### CPU
+
+On any machine with PyTorch/XLA installed, you can run our MNIST example on CPU
+like this:
+
+```
+PJRT_DEVICE=CPU python3 xla/test/test_train_mp_mnist.py --fake_data
+```
+
+### v4 TPU
+
+To create a new TPU with PyTorch/XLA r1.13 installed:
+
+```
+gcloud alpha compute tpus tpu-vm create $USER-pjrt --accelerator-type=v4-8 --version=tpu-vm-v4-pt-1.13 --zone=us-central2-b --project=$PROJECT
+```
+
+On a v4-8, you can run our ResNet50 example like this:
+
+```
+git clone --depth=1 --branch r1.13 https://github.com/pytorch/xla.git
+PJRT_DEVICE=TPU python3 xla/test/test_train_mp_imagenet.py --fake_data --batch_size=256 --num_epochs=1
+```
+
+By default, PjRt will use all TPU chips. To use only one TPU chip, configure
+`TPU_PROCESS_BOUNDS` and `TPU_VISIBLE_CHIPS`:
+
+```
+TPU_PROCESS_BOUNDS=1,1,1 TPU_VISIBLE_CHIPS=0 PJRT_DEVICE=TPU python3 xla/test/test_train_mp_imagenet.py --fake_data --batch_size=256 --num_epochs=1
+```
+
+#### Pods
+
+On TPU Pods, use `gcloud` to run your command on each TPU in parallel:
+
+```
+gcloud alpha compute tpus tpu-vm ssh $USER-pjrt --zone=us-central2-b --project=$PROJECT --worker=all --command="git clone --depth=1 --branch r1.13 https://github.com/pytorch/xla.git"
+gcloud alpha compute tpus tpu-vm ssh $USER-pjrt --zone=us-central2-b --project=$PROJECT --worker=all --command="PJRT_DEVICE=TPU python3 xla/test/test_train_mp_imagenet.py --fake_data --batch_size=256 --num_epochs=1"
+```
+
+### GPU
+
+Coming soon in a future release!
+
+## Key differences from XRT
+
+Although in most cases we expect PjRt and XRT to work mostly interchangeably
+from the end-user's perspective (especially on TPU v4), there are some subtle
+differences that are important to keep in mind. Importantly, XRT was designed
+around the TPU Node architecture, so it will always spawn a client and a server
+process, even on TPU VMs. Thus, every batch of inputs has additional latency
+from serializing and deserializing data.
+
+PjRt uses the local device directly with no intermediate server process. In the
+default configuration, PjRt will create one process per TPU chip, or 4 processes
+per TPU host. See the [Cloud TPU documentation](https://cloud.google.com/tpu/docs/system-architecture-tpu-vm)
+for more information about TPU architecture.
+
+- Performance gains are possible for workloads constrained by data transfer
+  speeds.
+- Under XRT, the server process is the only process that interacts with the TPU
+  devices, and client processes don't have direct access to the TPU devices.
+  When profiling a single-host TPU (e.g. v3-8 or v4-8), you would normally see 8
+  device traces (one for each TPU core). With PjRt, each process has one chip,
+  and a profile from that process will show only 2 TPU cores.
```

Comment:
on v3 right? I thought we are only going to see 1 device trace on v4.

Reply:
I can't find any screenshots, but I'm pretty sure that v4 also shows both cores in the device trace.